bots/mainbot/castle.js

```diff
@@ -8,379 +8,163 @@ import combat from './combat.js'
+        if(self.map.length <= 40 || self.castle_count === 1){
+            self.strategy = 'crusaderRush'
+        }
+        else{
+            self.strategy = 'prophetArmy'
+        }
+        self.builtItems = 0;
+
+        if(self.castle_count === 1){
+            // IF only one castle set the queuePos to 0.
+            // Anything mod the
+            self.queue = [self.id];
+        }
+        else{
+            // Sort by ids
+            let castleIds = []
+            for(let i = 0; i < visible.length; ++i){
+                castleIds.push(visible[i].id)
+            }
+            self.queue = castleIds.sort(function(a, b){return a - b})
+            return
+        }
 	}
+    if( self.queue[(self.builtItems % self.castle_count)] === self.id){
+        return castle.buildUnits(self);
+    }
+    else{
+        return undefined
+    }
 
-/*	if(self.me['turn'] == 2 )
-	{
-			var d = getBuildDir(self.me);
-			if (!(d === undefined)){
-			self.log('Building a pilgrim at ' + (self.me.x+1) + ',' + (self.me.y+1));
-			pilgrims.number++;
-			return self.buildUnit(SPECS.PILGRIM, d.x, d.y);
-		}
-
-	}
-
-	if(self.me['turn'] == 3)
-	{
-		self.log('Turn 2 for blue team')
-				var d = getBuildDir(self.me);
-				if (!(d === undefined)){
-					self.log('Building a prophet at ' + (self.me.x+1) + ',' + (self.me.y+1));
-					 return self.buildUnit(SPECS.PROPHET, d.x, d.y);
-				}
-
-			var d = getBuildDir(self.me);
-			if (!(d === undefined)){
-			self.log('Building a pilgrim at ' + (self.me.x+1) + ',' + (self.me.y+1));
-			pilgrims.number++;
-			return self.buildUnit(SPECS.PILGRIM, d.x, d.y);
-		}
-} */
-
-
-
-/*	if (self.me.turn > 5 && self.me.turn % 2 == 0 && self.karbonite > 30 && self.fuel > 150 && Math.random() < .3333)
-	   {
-		   var d = getBuildDir(self.me);
-	   if (!(d === undefined)){
-		   self.log('Building a prophet at(blue team) after 100 turn ' + (self.me.x+1) + ',' + (self.me.y+1));
-		   pilgrims.number++;
-		   return self.buildUnit(SPECS.PROPHET, d.x, d.y);
-		   }
-	   }
-
-	   if ( self.me.turn >5 && self.me.turn % 2 != 0 && self.karbonite > 30 && self.fuel > 150 && Math.random() < .3333)
-	   {
-		   var d = getBuildDir(self.me);
-		   if (!(d === undefined)){
-			   self.log('Building a crusader at(blue team)  mod 10 ' + (self.me.x+1) + ',' + (self.me.y+1));
-			   pilgrims.number++;
-			   return self.buildUnit(SPECS.CRUSADER, d.x, d.y);
-			   }
-	   }   */
+};
 
-	 /*  var enemies = utilities.enemiesInRange(self);
-	   if(enemies.length > 0){
-		   for(let i = 0; i < enemies.length; ++i){
-	   //	if(enemies[i].unit === SPECS.CASTLE){
-		  // 		self.log("Attacking Castle");
-				   return combat.attackBot(self, enemies[i]);
-			   }
-		   } */
+// Lets us split up the function for building. If it's not this castle's turn we
+// Can skip all of the build logic
+castle.buildUnits = (self) => {
+    switch (self.strategy) {
+        case 'crusaderRush':
+            return castle.crusaderRush(self);
+            break;
```

Comment:
break statements are redundant because the previous statement returns from the function